tree.go

```diff
@@ -411,6 +418,21 @@ walk: // Outer loop for walking the tree
 				idxc := path[0]
 				for i, c := range []byte(n.indices) {
 					if c == idxc {
+						if strings.HasPrefix(n.children[len(n.children)-1].path, ":") {
+							skipped = &skip{
+								path: prefix + path,
+								paramNode: &node{
```

Comment:
why not just make this data structure take `n`?

Reply:
Because this operation removes indices to prevent prefix checking.